### ISSUE TITLE
HOW TO ?  query by boolean 

### ISSUE THREAD

**Issue by Shahar2k5** [pardom-zz/ActiveAndroid#166]
hey guys,

i need to run select by boolean, is my query ok?

@Table(name = "MyMessages")
public class MyMessage extends Model {
...
...
    @Column(name = "read")
    public boolean read;
...
...
}

return new Select().from(MyMessage.class).where("read = ?", false).execute();

get all messages return good result, the query above returns count 0.

Thanks.

Shahar.

**Comment by joshuapinter** [comment]
Looks right. Have you tried it?

**Comment by Shahar2k5** [comment]
Didnt work.

I tried with '=0'
Seems ok, gonna try with '=1' tomorow at work to see if thats the case..

**Comment by joshuapinter** [comment]
You're right. I didn't like how the Booleans and Integers were handled so I essentially store all of my columns as Strings and deal with them outside of ActiveAndroid.

I found that Booleans didn't work right and Integers would always be set to 0 even if the actual value was 'null'. 'null' and 0 mean different things in our world and until I have time to patch AA to handle them better, sticking with Strings worked best for me.

**Comment by leonardoxh** [comment]
Booleans are serialized into integers by AA.

when you need fetch the data on where use integer comparators for booleans 0 means false and 1 true.

**Comment by joshuapinter** [comment]
@leonardoxh I would exercise caution when using Booleans and Integers in AA as I found `null` values get lost.

**Comment by sagarjs** [comment]
AA should serialize the parameters as well so that boolean in parameter gets automatically serialized to 0 or 1 when the query is generated
